### PR TITLE
[incubator/kubeless] Update kubeless version and fix resource issues

### PR DESCRIPTION
**Pull request by d0x2f**
#### Is this a new chart

No

#### What this PR does / why we need it:

Updates the version of the kubeless function controller, fixes invalid resource definitions due to kube api and updated CRD handling for helm3 compatibility.

#### Which issue this PR fixes

None

#### Special notes for your reviewer:

The description field on CRDs isn't valid any more and deployment resources live in the apps/v1 api now.
They were all the changes I intended to make, but after the e2e tests failed I needed to include a change to support helm 3 wrt the crds directory.
In helm 3, CRDs in that dir no longer support templating.
This meant that the `kubeless.io/namespace` annotation had to be removed from the functions.kubeless.io CRD.
Because of this the function manager would no longer be able to read the target namespace from the CRD and would default to "kubeless" which may be wrong depending on where the chart was installed.
To fix this I added the KUBELESS_NAMESPACE env var to each container to inform the controllers of the correct namespace.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: incubator/kubeless/crds/cronjobtrigger-crd.yaml [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-description: CRD object for Cron trigger type
 kind: CustomResourceDefinition
 metadata:
   name: cronjobtriggers.kubeless.io
   labels:
-{{ include "labels.standard" . | indent 4 }}
+    app: kubeless
+  annotations:
+    helm.sh/hook: crd-install
+    helm.sh/hook-delete-policy: before-hook-creation
```
Comment (andresmgot) [inline incubator/kubeless/crds/cronjobtrigger-crd.yaml:9]: why are you adding the delete-policy? Also, I think that the `crd-install` annotation is no longer needed.

File: incubator/kubeless/templates/crds.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{{- range $path, $bytes := .Files.Glob "crds/*.yaml" }}
+{{ $.Files.Get $path }}
+---
+{{- end }}
```
Comment (andresmgot) [inline incubator/kubeless/templates/crds.yaml:4]: is this file needed?

File: incubator/kubeless/values.yaml [hunk 3]
```diff
@@ -10,7 +10,7 @@ controller:
     functionController:
       image:
         repository: kubeless/function-controller
-        tag: v1.0.4
+        tag: v1.0.5
```
Comment (andresmgot) [inline incubator/kubeless/values.yaml:13]: Can you update the versions of the http and cronjob trigger controllers as well? (The latest ones are `v1.0.1`)

**Review by andresmgot — COMMENTED** [review]
Thanks @d0x2f for updating the chart! I have a couple of questions and a small request. It's a pity that we are losing the templating the CRDs but I think there is nothing we can do there.

File: incubator/kubeless/crds/cronjobtrigger-crd.yaml (hunk 1 above)
Comment (d0x2f) [reply incubator/kubeless/crds/cronjobtrigger-crd.yaml:9]: You're right, but this combined with the crds.yaml file is a temporary hack so that the chart can support both helm 2 and 3.

See #19008, I'm guessing kubeless was missed because it was in the incubator repo rather than stable.

Alternatively we could require a minimum helm version of 3 for the chart, but I think this repo aims to support both until EOL for helm 2.

File: incubator/kubeless/templates/crds.yaml (hunk 2 above)
Comment (d0x2f) [reply incubator/kubeless/templates/crds.yaml:4]: See previous comment.

File: incubator/kubeless/values.yaml (hunk 3 above)
Comment (d0x2f) [reply incubator/kubeless/values.yaml:13]: I noticed that the v1.0.1 tags aren't available in the [bitnami namespace](https://hub.docker.com/r/bitnami/http-trigger-controller/tags) in dockerhub that's used in this chart. It's present in the [kubeless namespace](https://hub.docker.com/r/kubeless/http-trigger-controller/tags) however, should I switch the images to that?

Comment (andresmgot) [reply incubator/kubeless/values.yaml:13]: ah, I didn't notice that these were bitnami images. Yes. please, use the `kubeless` namespace instead.

File: incubator/kubeless/crds/cronjobtrigger-crd.yaml (hunk 1 above)
Comment (andresmgot) [reply incubator/kubeless/crds/cronjobtrigger-crd.yaml:9]: :+1: got it

File: incubator/kubeless/values.yaml (hunk 3 above)
Comment (d0x2f) [reply incubator/kubeless/values.yaml:13]: Ok @andresmgot , that's been updated.

**Review by andresmgot — APPROVED** [review]
Great, thanks for the changes!
